Multisig/Data/Services/Notification Service/RegisterNotificationTokenRequest.swift

```diff
@@ -7,27 +7,64 @@
 //
 
 import Foundation
+import Web3
 
 struct RegisterNotificationTokenRequest: JSONRequest {
-    let uuid: String?
+    let uuid: String
     let safes: [String]
     let cloudMessagingToken: String
     let bundle: String
     let version: String
     let deviceType: String = "IOS"
     let buildNumber: String
+    let timestamp: String?
+    let signatures: [String]?
+
     var httpMethod: String { return "POST" }
     var urlPath: String { return "/api/v1/notifications/devices/" }
 
     typealias ResponseType = Response
 
-    init(deviceID: UUID? = nil, safes: [Address], token: String, bundle: String, version: String, buildNumber: String) {
-        self.uuid = deviceID?.uuidString.lowercased()
+    init(deviceID: String,
+         safes: [Address],
+         token: String,
+         bundle: String,
+         version: String,
+         buildNumber: String,
+         timestamp: String?) throws {
+
+        guard UUID(uuidString: deviceID) != nil else {
+            throw "'deviceID' should be UUID string"
```

Comment:
I would make it preconditionFailure() or just precondition instead of guard because this is developer error

Reply:
ok